build/dockerfiles/Dockerfile

```
@@ -0,0 +1,71 @@
+FROM debian:bullseye@sha256:71cb300d5448af821aedfe63afd55ba05f45a6a79f00dcd131b96b780bb99fe4
+
+# Set system time to 00:00:01 UTC
+ENV SOURCE_DATE_EPOCH=1
+
+# Avoid prompts during installation
+ENV DEBIAN_FRONTEND=noninteractive
+
+# Install curl
+RUN apt-get update && apt-get install -y curl jq
+
+# Copy apt install files from repo
+COPY apt/packages/ ./
+
+# Copy digest for apt install files
+COPY apt/package_urls.txt ./
+
+# File containing the list of packages and hashes
+ENV PACKAGE_LIST_FILE="./package_urls.txt"
+
+# Download install files, check hashes, and install
+RUN while IFS= read -r line; do \
+        package_url=$(echo "$line" | awk '{print $1}'); \
+        package_hash=$(echo "$line" | awk '{print $2}'); \
+        package_name=$(basename "$package_url"); \
+        if [ ! -f "./$package_name" ]; then \
+            echo "Downloading $package_url"; \
+            curl -O "$package_url"; \
+        fi; \
+        echo "Verifying hash for $package_name"; \
+        echo "$package_hash *$package_name" > temp_checksum_file; \
+        sha256sum -c temp_checksum_file; \
+        if [ $? -ne 0 ]; then \
+            echo "Hash check failed for $package_name"; \
+            exit 1; \
+        fi; \
+    done < "$PACKAGE_LIST_FILE" \
+    && dpkg -i ./*.deb
+
+# Create nimiq directory for build
+RUN mkdir -p /home/nimiq/
+
+# Change homedir
+WORKDIR /home/nimiq
+
+# Fetch the latest git tag and download the corresponding source code archive
+RUN CORE_RS_ALBATROSS_TAG=$(curl -s "https://api.github.com/repos/nimiq/core-rs-albatross/tags" | jq -r '.[0].name') \
+    && curl -fLO "https://github.com/nimiq/core-rs-albatross/archive/${CORE_RS_ALBATROSS_TAG}.zip" \
+    && unzip "${CORE_RS_ALBATROSS_TAG}.zip" -d /home/nimiq \
+    && ls -la . \
+    && mv "/home/nimiq/core-rs-albatross-$(echo ${CORE_RS_ALBATROSS_TAG} | sed 's/^v//')" /home/nimiq/core-rs-albatross \
+    && rm "${CORE_RS_ALBATROSS_TAG}.zip"
+
+# Set workdir to source 
+WORKDIR /home/nimiq/core-rs-albatross
+
+# Install rustup
+RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.75.0
```

```suggestion
RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --default-toolchain 1.75.0 --profile minimal
```

No need to download clippy etc.